daemon/cmd/kube_proxy_replacement.go

```diff
@@ -54,16 +54,14 @@ func initKubeProxyReplacementOptions() (strict bool) {
 	}
 
 	if option.Config.KubeProxyReplacement == option.KubeProxyReplacementDisabled {
-		log.Infof("Auto-disabling %q, %q, %q, %q, %q features",
+		log.Infof("Auto-disabling %q, %q, %q, %q features",
 			option.EnableNodePort, option.EnableExternalIPs,
-			option.EnableHostReachableServices, option.EnableHostPort,
-			option.EnableSessionAffinity)
+			option.EnableHostReachableServices, option.EnableHostPort)
 
 		disableNodePort()
 		option.Config.EnableHostReachableServices = false
 		option.Config.EnableHostServicesTCP = false
 		option.Config.EnableHostServicesUDP = false
-		option.Config.EnableSessionAffinity = false
```

Comment:
Overall looks reasonable to me. I think we never considered extending this for the 'legacy' kube-proxy case, but makes sense to have this for ClusterIP there. The change would break with older kernels that do not have LRU map though, that still needs some minor tweaking in your PR, see ...
```
	if option.Config.EnableSessionAffinity {
		if !probesManager.GetMapTypes().HaveLruHashMapType {
			msg := "SessionAffinity feature requires BPF LRU maps"
			if strict {
				log.Fatal(msg)
			} else {
				log.Warnf("%s. Disabling the feature.", msg)
				option.Config.EnableSessionAffinity = false
			}
		}
	}
```
... where this would fatal given strict is set.

Reply:
What behavior do you expect to have here? I kind of prefer the current way where it fatals when a feature enabled by users can't be turned on. Note this flag is false by default.

Reply:
I can remove the log.Fatal if that's what you want.

Reply:
Re log.Fatal: I think it's fine as it is, as running with `--kube-proxy-replacement=disabled` the `strict` is always false, so we would never fatal.

Reply:
When `--kube-proxy-replacement=disabled`, `strict` is also true:

```
strict = option.Config.KubeProxyReplacement != option.KubeProxyReplacementProbe
```

While I'd like to enforce this feature if people enable it specifically, it will cause crash for those running v4.9.17 - v4.10. We may want to remove this log.Fatal to avoid such crash. For this PR, I simply want people to have a way to enable it in kube-proxy mode.

Reply:
Yeah, but when `--kube-proxy-replacement=disabled`, then the init function will return before the strict checks happen: https://github.com/cilium/cilium/blob/master/daemon/cmd/kube_proxy_replacement.go#L68.

Reply:
Added a section to disable session-affinity when `--kube-proxy-replacement=disabled`